### PR TITLE
serde: implement `serialize_dict`

### PR DESCRIPTION
**Pull request by tcoratger**

### CODE REVIEW

File: crates/exex/src/serde.rs [hunk 1]
```diff
@@ -1254,4 +1345,209 @@ mod tests {
...
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 6usize).unwrap(), key3.clone())
+            .expect("failed to insert key2");
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 8usize).unwrap(), value3.clone())
+            .expect("failed to insert value2");
```
Comment (ClementWalter) [inline crates/exex/src/serde.rs:1418]: insert directly a `Vec`, no need to do this 6 times

File: crates/exex/src/serde.rs [hunk 2]
```diff
@@ -1254,4 +1345,209 @@ mod tests {
...
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 0usize).unwrap(), key.clone())
+            .expect("failed to insert key");
+        kakarot_serde
+            .runner
+            .vm
+            .insert_value((dict_ptr + 2usize).unwrap(), null_value)
+            .expect("failed to insert null value");
```
Comment (ClementWalter) [inline crates/exex/src/serde.rs:1451]: see above

File: crates/exex/src/serde.rs (hunk 1 above)
Comment (tcoratger) [reply crates/exex/src/serde.rs]: ah yes forgot this trick again, fixed

File: crates/exex/src/serde.rs (hunk 2 above)
Comment (tcoratger) [reply crates/exex/src/serde.rs]: fixed

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**